### PR TITLE
Move `cometd-javascript-common` to EcmaScript Modules

### PR DESCRIPTION
**Pull request by ghoullier**
# Purpose

https://github.com/cometd/cometd/issues/1511

## Changes

This PR split cometd.js file into multiples files with EcmaScript Modules.

I did not convert prototype based class to es6 class in this PR.

### CODE REVIEW

**Comment by ghoullier** [comment]
@sbordet I'm not familiar with maven, but it seems the toolchain does not supports ESModules.

**Comment by sbordet** [comment]
@ghoullier sorry for the delay handling this PR, been busy.

I'll look as soon as possible. Thanks!

**Comment by ghoullier** [comment]
@sbordet do you had the time to check the PR?

**Comment by sbordet** [comment]
@ghoullier next in my todo list this week and the next (with easter in between). Thanks!

File: cometd-javascript/cometd-javascript-common/src/main/webapp/js/cometd/index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  CallbackPollingTransport,
+  WebSocketTransport,
+  Utils,
+  Z85,
+  AckExtension,
+  BinaryExtension,
+  ReloadExtension,
+  TimeStampExtension,
+  TimeSyncExtension,
+};
```
Comment (ghoullier) [inline cometd-javascript/cometd-javascript-common/src/main/webapp/js/cometd/index.js:30]: We also have this syntax 👇
```suggestion
export { CometD } from "./CometD.js";
export { Transport } from "./Transport.js";
export { RequestTransport } from "./RequestTransport.js";
export { LongPollingTransport } from "./LongPollingTransport.js";
export { CallbackPollingTransport } from "./CallbackPollingTransport.js";
export { WebSocketTransport } from "./WebSocketTransport.js";
export { Utils } from "./Utils.js";
export { Z85 } from "./Z85.js";

export { AckExtension } from "./AckExtension.js";
export { BinaryExtension } from "./BinaryExtension.js";
export { ReloadExtension } from "./ReloadExtension.js";
export { TimeStampExtension } from "./TimeStampExtension.js";
export { TimeSyncExtension } from "./TimeSyncExtension.js";
```

**Comment by sbordet** [comment]
@ghoullier I'm going to accept this contribution, but I will rework it to comply with the project goals.
In particular, I will restore the license header, the indentation, and reduce the splitting of files.
Thanks!

**Comment by sbordet** [comment]
Closing this PR, since it has been integrated.

@ghoullier thanks for pushing us doing this work.
We have extended it to the demos and examples, and the test suite.

See also: #1714.

**Pull request closed without merging**